tests/Functional/MemoizeTest.php

```diff
@@ -195,6 +195,20 @@ public function testResetByPassingNullAsCallable()
         memoize([$this->callback, 'execute']);
     }
 
+    public function testUsingDifferentCallbackWithSameArgument()
+    {
+        $callbackHello = function(string $name): string {
+            return 'Hello ' . $name;
+        };
+
+        $callbackHey = function(string $name): string {
+            return 'Hello ' . $name;
```

Comment:
It also says "Hello" here, does that matter?

Reply:
Typo!

I could not reproduce our bug.